packages/in_app_purchase/in_app_purchase_android/lib/src/types/google_play_product_details.dart

```diff
@@ -11,35 +11,135 @@ import '../../billing_client_wrappers.dart';
-      title: skuDetails.title,
-      description: skuDetails.description,
-      price: skuDetails.price,
-      rawPrice: skuDetails.priceAmountMicros / 1000000.0,
-      currencyCode: skuDetails.priceCurrencyCode,
-      currencySymbol: skuDetails.priceCurrencySymbol,
-      skuDetails: skuDetails,
+    assert(productDetails.productType == ProductType.inapp);
+    assert(productDetails.oneTimePurchaseOfferDetails != null);
+
+    final OneTimePurchaseOfferDetailsWrapper oneTimePurchaseOfferDetails =
+        productDetails.oneTimePurchaseOfferDetails!;
+
+    final String formattedPrice = oneTimePurchaseOfferDetails.formattedPrice;
+    final double rawPrice =
+        oneTimePurchaseOfferDetails.priceAmountMicros / 1000000.0;
+    final String currencyCode = oneTimePurchaseOfferDetails.priceCurrencyCode;
+    final String currencySymbol =
+        formattedPrice.isEmpty ? currencyCode : formattedPrice[0];
+
+    return GooglePlayProductDetails._(
+      id: productDetails.productId,
+      title: productDetails.title,
+      description: productDetails.description,
+      price: formattedPrice,
+      rawPrice: rawPrice,
+      currencyCode: currencyCode,
+      currencySymbol: currencySymbol,
+      productDetails: productDetails,
     );
   }
 
-  /// Points back to the [SkuDetailsWrapper] object that was used to generate
-  /// this [GooglePlayProductDetails] object.
-  final SkuDetailsWrapper skuDetails;
+  /// Generate a [GooglePlayProductDetails] object based on an Android
+  /// [ProductDetailsWrapper] object for a subscription product.
+  /// Subscriptions can consist of multiple base plans, and base plans in turn
+  /// can consist of multiple offers. This method generates a list where every
+  /// element corresponds to a base plan or its offer.
+  ///
+  /// Subscriptions can consist of multiple base plans, and base plans in turn
+  /// can consist of multiple offers. [subscriptionIndex] points to the index of
+  /// [productDetails.subscriptionOfferDetails] for which the
+  /// [GooglePlayProductDetails] is constructed.
+  factory GooglePlayProductDetails._fromSubscription(
+    ProductDetailsWrapper productDetails,
+    int subscriptionIndex,
+  ) {
+    assert(productDetails.productType == ProductType.subs);
+    assert(productDetails.subscriptionOfferDetails != null);
+    assert(subscriptionIndex < productDetails.subscriptionOfferDetails!.length);
+
+    final SubscriptionOfferDetailsWrapper subscriptionOfferDetails =
+        productDetails.subscriptionOfferDetails![subscriptionIndex];
+
+    final PricingPhaseWrapper firstPricingPhase =
+        subscriptionOfferDetails.pricingPhases.first;
+    final String formattedPrice = firstPricingPhase.formattedPrice;
+    final double rawPrice = (firstPricingPhase.priceAmountMicros) / 1000000.0;
+    final String currencyCode = firstPricingPhase.priceCurrencyCode;
+    final String currencySymbol =
+        formattedPrice.isEmpty ? currencyCode : formattedPrice[0];
```

Comment:
Same question here; is this actually safe?

Extracting a helper method, with comments about why this is the logic, would be helpful.